src/store/store.js

```diff
@@ -57,6 +57,7 @@ const saveStateToLocalStorage = (state) => {
     theme: state.theme,
     isPrettyName: state.isPrettyName,
     showFeatureHints: state.showFeatureHints,
+    showDatasetPreviews: state.showDatasetPreviews,
```

Comment:
Hi @SajidAlamQB, 

There is an issue setting `showDatasetPreviews` this way. For other flags like `showFeatureHints` we are not sending a request to backend for updating the toggleStatus. So either having the default or reading from localStorage works. 

But for this specific flag, we need to get the initial status from the backend. This would require you to create another api `/preferences` which will be a GET request. This way the FE and backend will be in sync even when the browser is changed or the localStorage is deleted. Let me know if this is not clear. Thank you

Reply:
Could you elaborate on this a bit more, don't we assume that `showDatasetPreviews` is enabled by default?

Reply:
Let's connect and discuss